CoreFoundation/RunLoop.subproj/CFRunLoop.c

```diff
@@ -95,7 +97,7 @@ DISPATCH_EXPORT void _dispatch_main_queue_callback_4CF(void);
 #include <sys/eventfd.h>
 #include <sys/timerfd.h>
 
-dispatch_runloop_handle_t _dispatch_get_main_queue_handle_4CF(void);
```

Comment:
What is this necessary for? Why did we compile earlier without and can't now?

Reply:
I believe that it may be an artifact of my configuration, but, include ordering is also worrisome.  This requires a private header from libdispatch (`private/private.h`) which is attempted to be included below.  I can try to dig through that a bit more, but I thought that desugaring was better than trying to get through to the private headers (or is there a nice approach that apple has for getting to private headers for SPI?)

Reply:
Can we move the include up?

Reply:
@parkera, sure, I don't think that should result in anything breaking.  It does require a bit more work to move around as it was embedded in a weird way (or maybe it was the time of day).

Reply:
@parkera, @millenomi - so, I looked into this when I wasn't half asleep.  So, the problem is that the header that it is using (`dispatch/private.h`) is a private header (I know, shocking given the name of the file).  This is possible to install via the `INSTALL_PRIVATE_HEADERS` option.  However, it seems bad to have to install the private headers if you are sharing the libdispatch build (it will get installed into the toolchain image).  We *could* add the libdispatch source path to the search order as a workaround if you don't want to desugar the type.

Reply:
Hm, this is the only thing from dispatch's private headers that we use in the whole project?

Reply:
Yeah, it was the only thing that I hit that needed the private header explicitly.  The other things are declared and don't have any sugar that is not replicated here.

Reply:
@parkera, @millenomi - so what is your opinion on this?  The three options that I see (in reverse order of [my] preference):
1) require private headers to always be installed for swift
2) require path to libdispatch sources to get the private header
3) desugar the type

Reply:
I could imagine a `has_include` style check here, perhaps redefining the type if the private header is not available?